tests/input-format/src/format.rs

```diff
@@ -0,0 +1,47 @@
+use serde::{Deserialize, Serialize};
+
+/// Position of a mouse cursor on the screen.
+///
+/// Mouse cursor positions are sized relative to the Flash stage's dimensions,
+/// regardless of the native window's size or pixel density. For example, a
+/// (640x480) stage movie on a 2x display (on platforms that report physical
+/// pixels) or at 2x the size will see mouse clicks at its bottom right corner
+/// on (1280x960), relative to the window. That coordinate needs to be scaled
+/// down to match the desired stage.
+#[derive(Serialize, Deserialize, Debug)]
+pub struct MousePosition(pub f64, pub f64);
+
+/// Which mouse button is being pressed or released.
+#[derive(Serialize, Deserialize, Debug, Copy, Clone)]
+pub enum MouseButton {
+    Left,
+    Middle,
+    Right,
+}
+
+/// All automated event types supported by FlashTAS.
+///
+/// A FlashTAS input file consists of a string of `AutomatedEvent`s which are
+/// played back by FlashTAS.
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(tag = "type")]
+pub enum AutomatedEvent {
+    /// End the current frame's input and wait for the next frame before
+    /// continuing to inject input.
+    Wait,
+
+    /// Move the mouse to a new cursor position.
+    MouseMove { pos: MousePosition },
+
+    /// Click a mouse button.
+    MouseDown {
+        pos: MousePosition,
+        btn: MouseButton,
+    },
+
+    /// Release a mouse button.
+    MouseUp {
+        pos: MousePosition,
+        btn: MouseButton,
+    },
+}
```

Comment:
TODO maybe someday: It'd be nice if we could reuse some of the `core` types here. At least things like `MouseButton` and `KeyCode`, or possibly this could be all of `PlayerEvent`.

Naturally I wouldn't want to require all of `ruffle_core` here for a few tiny types, so maybe we can keep this in mind if we ever reorganize our workspace structure. On the other hand, having the separation might be a bit nicer to prevent the input format breakage if we add things to the `core` side structs.

Reply:
There's a few subtle things that need to be different from `PlayerEvent`, too; while it might be justifiable to add `serde` impls for it, I doubt a `Wait` event would make sense outside of automated contexts.